kartothek/io_components/read.py

```diff
@@ -108,16 +107,20 @@ def dispatch_metapartitions_from_factory(
         if isinstance(ix, ExplicitSecondaryIndex)
     }
 
-    if dispatch_by:
+    if dispatch_by is not None:
         base_df = cast(pd.DataFrame, base_df)
 
-        # Group the resulting MetaParitions by partition keys or a subset of those keys
-        merged_partitions = base_df.groupby(
-            by=list(dispatch_by), sort=True, as_index=False
-        )
+        if dispatch_by == []:
```

Comment:
I would prefer `if dispatch_by` or `if len(dispatch_by) == 0`. I don't trust the comparison to a mutable object, even though it is empty. The `if dispatch_by` is probably considered the most "pythonic" way

Reply:
Here I wanted to actively differentiate from the `dispatch_by=None` case (for readability). However `if len(dispatch_by) == 0` is much more elegant.